### PR TITLE
latexify hmm_poisson_changepoint.ipynb

### PR DESCRIPTION
**Pull request by TakshPanchal**
### Figure Number

Figure 29.6, 29.7 and 29.8 
#### Figure 29.6
| Before PR | After PR |
| :-: | :-: |
|29.6B|29.6A

#### Figure 29.7
| Before PR | After PR |
| :-: | :-: |
|Screenshot from 2022-07-13 15-31-57|Screenshot from 2022-07-13 15-31-41

#### Figure 29.8
| Before PR | After PR |
| :-: | :-: |
|Screenshot from 2022-07-13 15-37-24|Screenshot from 2022-07-13 15-37-08|

### Issue 
#113 

## Steps

- [X] Latexify

## Checklist:

- [X] Performed a self-review of the code
- [X] Tested on Google Colab.

## Potential problems/Important remarks
When I tried to add legends in fig 29.7, it was too cumbersome. Should I add a legend to fig 29.7 figures?

### CODE REVIEW

**Comment by karm-patel** [comment]
Regarding the legend in Fig 29.7
1. In the case of the same legend in every plot, we usually add a legend in only one plot, the ideally first plot should have a legend, but if the first plot doesn't have space to put a legend, then we can choose any plot which has more space.
2. In case if legend's boundary is overlapping with plot you can use `ax.legend(..., frameon=False)`

**Comment by karm-patel** [comment]
### Fig 29.6
1. You can put the title of the 2nd subplot in subcaption like (b) Inferred
2. Because we need the same `ylim` for both the subplots, the first comment would resolve this issue.

**Comment by karm-patel** [comment]
@TakshPanchal , Other things look good to me, you can create a PR in the main repo after resolving the above comments.

**Comment by TakshPanchal** [comment]
Removing the frame still didn't work out so I set `fontsize` to `xx-small`
image

**Comment by TakshPanchal** [comment]
## Fig 29.6
image

**Comment by TakshPanchal** [comment]
> * You can put the title of the 2nd subplot in subcaption like (b) Inferred

It will be the change in the latex file of the book right? :thinking:

**Comment by karm-patel** [comment]
> Removing the frame still didn't work out so I set `fontsize` to `xx-small` image

Try `fontsize`  in numbers always, so we can get an idea, start with 9 and then decrease, ideally 7 should be minimum
Here you can apply some hacks, like add "\n" - plt.plot(..., label="inferred\nrate")

**Comment by TakshPanchal** [comment]
Okay. understood.

**Comment by TakshPanchal** [comment]
How's this looking?
image

**Comment by karm-patel** [comment]
> How's this looking? image

Ummm, looks a little crowdy,
Maybe add a legend in 4th plot. Because 4th plot is also the first plot. And please tell me `fontsize` also

**Comment by karm-patel** [comment]
One more thing, Now  I think you don't need to use reverse style legend labels, right?

**Comment by karm-patel** [comment]
@TakshPanchal 
try out this comment also :)
image

**Comment by TakshPanchal** [comment]
Done :)
Screenshot from 2022-07-14 19-32-19

**Comment by karm-patel** [comment]
Great @TakshPanchal , Go ahead with PR in the main repo.

**Comment by TakshPanchal** [comment]
Done :) and `fontsize` is 7
Screenshot from 2022-07-14 19-32-19

**Comment by karm-patel** [comment]
@TakshPanchal Wait,

Do you know the reason why this low spike is there? Previously it was not there
image

File: notebooks/book2/29/hmm_poisson_changepoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,1013 @@
+{
```
Comment (karm-patel) [inline notebooks/book2/29/hmm_poisson_changepoint.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    rng_key, rng_normal = jax.random.split(rng_key)
</pre><p>may be you can try changing <code>rng_key</code> to reproduce the results. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/129/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

**Comment by karm-patel** [comment]
@TakshPanchal, don't spend much time to reproduce 2nd plot, just try some rng_keys, otherwise, you can ask this to Dr. Kevin while creating PR, who knows this spike is obvious.

**Comment by TakshPanchal** [comment]
Okay got it.

**Comment by TakshPanchal** [comment]
I tried some values of seeds and `rng_keys` but no change in spike.
